category/vm/compiler/ir/x86.cpp

```diff
@@ -331,9 +333,15 @@ namespace
             MONAD_VM_DEBUG_ASSERT(
                 remaining_base_gas >= instr.static_gas_cost());
             remaining_base_gas -= instr.static_gas_cost();
+            auto const result_bound = compute_result_bound(emit, instr);
             emit_instr<traits>(emit, instr, remaining_base_gas);
+            if (instr.stack_increase() > 0) {
```

Comment:
Nit: `instr.stack_increase()` is always 0 or 1, but type allows up to 256 while `compute_result_bound` assumes there's up to 1 result per instruction.